xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/Block.java

```diff
@@ -280,6 +281,62 @@ public interface Block extends Cloneable
      */
     void setParameters(Map<String, String> parameters);
 
+    /**
+     * @return all attributes
+     * @since 14.10.5
+     * @since 15.1RC1
+     */
+    @Unstable
+    default Map<String, Object> getAttributes()
```

Comment:
wdyt of not making those methods as `default` as they don't provide anything meaningful by default?

Reply:
This would be an API breakage and I wanted to avoid that, in particular as I need this change on the LTS branch for an important bug fix. Basically, what I implement by default is the behavior of an empty read-only attribute map.